homeassistant/components/media_player/frontier_silicon.py

```diff
@@ -0,0 +1,245 @@
+    import requests
+
+    if discovery_info is not None:
+        add_devices(
+            [FSAPIDevice(discovery_info, DEFAULT_PASSWORD)],
+            update_before_add=True)
+        return True
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        add_devices(
+            [FSAPIDevice(DEVICE_URL.format(host, port), password)],
+            update_before_add=True)
+        _LOGGER.debug('FSAPI device %s:%s -> %s', host, port, password)
+        return True
+    except requests.exceptions.RequestException:
+        _LOGGER.error('Could not add the FSAPI device at %s:%s -> %s',
+                      host, port, password)
+
+    return False
+
+
+class FSAPIDevice(MediaPlayerDevice):
+    """Representation of a Frontier Silicon device on the network."""
+
+    def __init__(self, device_url, password):
+        """Initialize the Frontier Silicon API device."""
+        self._device_url = device_url
+        self._password = password
+        self._state = STATE_UNKNOWN
+
+        self._name = None
+        self._title = None
+        self._mute = None
+        self._source = None
+        self._source_list = None
+        self._media_image_url = None
+
+    def get_fs(self):
+        """Create a fsapi session."""
+        from fsapi import FSAPI
+
+        return FSAPI(self._device_url, self._password)
+
+    # Properties
+    @property
+    def should_poll(self):
+        """Device should be polled."""
+        return True
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._name
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._title
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return SUPPORT_FRONTIER_SILICON
+
+    @property
+    def state(self):
+        """Return the state of the player."""
+        return self._state
+
+    # source
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._source_list
+
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self._source
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        return self._media_image_url
+
+    # Async actions
+
+    def update(self):
+        """Get the latest date and update device state."""
+        fs_device = self.get_fs()
+
+        if not self._name:
+            self._name = fs_device.friendly_name
+
+        if not self._source_list:
+            self._source_list = fs_device.mode_list
+
+        title = ''
+        artist = fs_device.play_info_artist
+        album = fs_device.play_info_album
+        info_name = fs_device.play_info_name
+        info_text = fs_device.play_info_text
+
+        if artist:
```

Comment:
Don't stuff everything into the title. Use the additional media fields as found here: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/media_player/__init__.py#L478